backend/pick-git/src/test/java/com/woowacourse/pickgit/integration/authentication/OAuthServiceIntegrationTest.java

```diff
@@ -22,12 +25,15 @@
 import org.springframework.boot.test.context.SpringBootTest.WebEnvironment;
 import org.springframework.boot.test.mock.mockito.MockBean;
 import org.springframework.context.annotation.Import;
+import org.springframework.test.annotation.DirtiesContext;
+import org.springframework.test.annotation.DirtiesContext.ClassMode;
 import org.springframework.test.context.ActiveProfiles;
 
 @Import(InfrastructureTestConfiguration.class)
 @DisplayName("OAuthService 통합 테스트 (UserRepository 사용)")
 @SpringBootTest(webEnvironment = WebEnvironment.NONE)
 @ActiveProfiles("test")
+@DirtiesContext(classMode = ClassMode.BEFORE_EACH_TEST_METHOD)
 class OAuthServiceIntegrationTest {
 
```

Comment:
```suggestion
@Import(InfrastructureTestConfiguration.class)
@DataJpaTest
@ActiveProfiles("test")
class OAuthServiceIntegrationTest {

    private static final String SECRET_KEY = "secret-key";
    private static final Long EXPIRATIONTIMEINMILLISECONDS = 3600000L;
    
    @MockBean
    private OAuthClient oAuthClient;

    private JwtTokenProvider jwtTokenProvider;

    private OAuthAccessTokenDao oAuthAccessTokenDao;

    @Autowired
    private UserRepository userRepository;

    private OAuthService oAuthService;

    @BeforeEach
    void setUp() {
        jwtTokenProvider = new JwtTokenProviderImpl(SECRET_KEY, EXPIRATIONTIMEINMILLISECONDS);
        oAuthAccessTokenDao = new CollectionOAuthAccessTokenDao();
        oAuthService = new OAuthService(oAuthClient, jwtTokenProvider, oAuthAccessTokenDao, userRepository);
    }
```
이부분은 굳이 `@SpringBootTest` 안쓰고, `@DataJpaTest`로 대체할 수 있을 것 같아요!

테스트 시간도 줄일 수 있어서 더 좋은 방안이지 않나 싶습니다!

Reply:
엄청 오묘하네... Integration test인데 Repository랑만 관련이 있다면 DataJpaTest로도 통합테스트가 가능하다니..